### DISCUSSION TITLE
Faster Javascript build times...

### DISCUSSION THREAD

**Discussion by richb-hanover** [PRQL/prql#2834] [General]
@max-sixty [Speaking of coming up with a cheap build system...](https://github.com/PRQL/prql/pull/2801#discussion_r1228575647) Is anyone looking at [Vite JS](https://vitejs.dev/) as an alternative to relying on webpack?

Vite uses ES Modules so the browser can load the Javascript files directly from the server instead of going through a (webpack) bundling step. It makes development extremely fast, and updating a Javascript file just requires that one file be reloaded instead of bundling the whole thing again (which takes 30-40 seconds on my Mac). Vite uses a bundler program (Rollup) to create production systems.

I am always wary of the _framework du jour_, especially in the Javascript world. But Vite has been around for quite a while and has real support. 

This is a long way of saying that sometime this summer I might be interested in converting PRQL to use Vite, unless someone else beats me to it.

**Comment by max-sixty** [comment]
I _think_ most of the build time comes from compiling the rust code on each run, rather than the JS. 

#1836 will mean we don't re-compile unless we need to.

Before changing the JS build, we should see how much time that's responsible for. It has worked reasonably well, and there is smaller bench of folks comfortable with JS if we do hit issues.

**Comment by richb-hanover** [comment]
OK. I'll sit tight 'til #1836 and other improvements come to pass. (It's not as if I will be be staring off to the horizon, wondering what to work on...) Thanks.
